tools/traffic/metrics/count_metric.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// CountMetric allows the count of a type of event to be tracked.
+type CountMetric interface {
+	Increment()
+}
+
+// countMetric a standard implementation of the CountMetric interface via prometheus.
+type countMetric struct {
+	metrics     *metrics
+	description string
+}
+
+// Increment increments the count of a type of event.
+func (metric *countMetric) Increment() {
+	metric.metrics.count.WithLabelValues(metric.description).Inc()
+}
+
+// NewCountMetric creates a new prometheus collector for counting metrics.
+func buildCounterCollector(namespace string, registry *prometheus.Registry) *prometheus.CounterVec {
+	return promauto.With(registry).NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "event_count",
```

Comment:
This way all counter will share the same name. Should this be configurable?

Reply:
Discussed this with Ian offline, he is satisfied with my explanation and gave me a 👍 to merge as is. Marking as resolved.